api-server/common/models/user.js

```diff
@@ -331,7 +335,8 @@ export default function(User) {
   });
 
   User.doesExist = function doesExist(username, email) {
-    if (!username && (!email || !isEmail(email))) {
+    username = username.toLowerCase();
```

Comment:
Operate on the property only after the validation check in the next line. This would throw an exception if the property is not valid.

Reply:
@raisedadead moved this down a statement